src/platforms/android/androidcontroller.cpp

```diff
@@ -162,14 +162,24 @@ void AndroidController::activate(const InterfaceConfig& config,
   jServer["publicKey"] = config.m_serverPublicKey;
   jServer["port"] = (double)config.m_serverPort;
 
-  // TODO: This could be done better with VpnService.Builder.excludeRoute()
-  // we just need a way to get this config down that far.
-  const auto localRoutes = Controller::getExcludedIPAddressRanges().flatten();
-  const auto fullAllowedIPs =
-      IPAddress::excludeAddresses(config.m_allowedIPAddressRanges, localRoutes);
+  // This could be done better with VpnService.Builder.excludeRoute()
```

Comment:
I actually went down the rabbit hole of implementing this with `VpnService.Builder.excludeRoute()` and it worked great, and only afterwards did I realize that it requires a higher API level than we ship with. So this implementation here is just crudely working around the issue, but in a way that doesn't require a change to our minimum Android version.